src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -122,6 +122,7 @@ public ValueTask DisposeAsync()
             static async ValueTask Await(int i, ValueTask vt, List<object> toDispose)
             {
                 await vt.ConfigureAwait(false);
+                i--;
```

Comment:
It may make sense to add a comment here explaining why `i` needs to be decremented here.

Reply:
Added